tests/S3/S3Transfer/S3TransferManagerTest.php

```diff
@@ -0,0 +1,3869 @@
+    ): S3Client
+    {
+        $index = 0;
+        return new S3Client([
+            'region' => 'eu-west-1',
+            'http_handler' => function (RequestInterface $request)
+            use ($bodyBuilder, $responses, &$index) {
+                $response = $responses[$index++];
+                if ($response['status'] < 400) {
+                    $headers = $response['headers'] ?? [];
+                    $body = call_user_func_array(
+                        $bodyBuilder,
+                            [
+                                $response['operation'],
+                                $response['body']
+                                ?? $response['contents']
+                                ?? null,
+                                &$headers
+                            ]
+                    );
+
+                    $this->parseCaseHeadersToAmzHeaders($headers);
+
+                    return new Response(
+                        $response['status'],
+                        $headers,
+                        $body
+                    );
+                } else {
+                    return new RejectedPromise(
+                        new S3TransferException(
+                            $response['errorMessage'] ?? ""
+                        )
+                    );
+                }
+            },
+        ]);
+    }
+
+    /**
+     * @param array $config
+     *
+     * @return void
+     */
+    private function parseConfigFromCamelCaseToSnakeCase(
+        array &$config
+    ): void
+    {
+        foreach ($config as $key => $value) {
+            // Searches for lowercaseUPPERCASE occurrences
+            // Then it is replaced by using group1_group2 found.
+            $newKey = strtolower(
+                preg_replace(
+                    "/([a-z0-9])([A-Z])/",
+                    "$1_$2",
+                    $key
+                )
+            );
+            unset($config[$key]);
+            $config[$newKey] = $value;
+        }
+    }
+
+    /**
+     * Checks if all words from the expected message appear in the actual message
+     * in the same order (allowing for gaps between words).
+     * Example that resolves to true:
+     *  expected: "error validating config"
+     *  actual:   "error validating download config"
+     *  reason: The words "error" -> "validating" -> "config" were found in the
+     *  same error in the actual message.
+     *
+     * Example that resolves to false:
+     *  expected: "error validating config"
+     *  actual:   "error in download config validation"
+     *  reason: The words error->validating->config were not
+     *  found in order.
+     *
+     * @param string $expectedMessage The message containing words to find
+     * @param string $actualMessage   The message to search within
+     *
+     * @return bool True if all expected words are found in order, false otherwise
+     */
+    private function assertEachWordMatchesTheErrorMessage(
```

Comment:
Are the error messages so different that `assertEqualsIgnoringCase` and a few trims or some adjustments to our error messages won't work?